lib/Sema/MiscDiagnostics.cpp

```diff
@@ -3990,6 +3990,59 @@ static void diagnoseDeprecatedWritableKeyPath(const Expr *E,
   const_cast<Expr *>(E)->walk(Walker);
 }
 
+static void maybeDiagnoseCallToKeyValueObserveMethod(const Expr *E,
+                                                     const DeclContext *DC) {
+  class KVOObserveCallWalker : public ASTWalker {
+    const ASTContext &C;
+
+  public:
+    KVOObserveCallWalker(ASTContext &ctx) : C(ctx) {}
+
+    void maybeDiagnoseCallExpr(const CallExpr *expr) {
```

Comment:
Probably not worth it to have `const` here to cast it out later in the walker...